src/Bladeburner/Types.ts

```diff
@@ -21,11 +14,12 @@ type AttemptFailure = { success?: undefined; message: string };
 export type Attempt<T extends object = object> = AttemptSuccess<T> | AttemptFailure;
 
 export type Action = Contract | Operation | BlackOperation | GeneralAction;
+export type TypeOfActionId<ActionType extends Action> = Pick<ActionType, "type" | "name">;
```

Comment:
This name sounds like it's literally `typeof ActionId`, which isn't accurate. I'd lean towards something like `ActionIdFor` or just `ActionId` itself.

Reply:
ActionId sounds good.